source/SpinalHDL/Design errors/out_of_range_constant.rst

```diff
@@ -0,0 +1,46 @@
+
+Out of Range Constant
+=====================
+
+Introduction
+------------
+
+SpinalHDL will check that in comparisons with literals the literal is not wider than the value compared to.
```

Comment:
```suggestion
SpinalHDL checks that in comparisons with literals the literal is not wider than the value compared to.
```

Prefer present of general truths.

Also is it only for comparisons?

Reply:
Good point about the comparisons. The error's thrown in `BinaryOperatorWidthableInput.checkLiteralRanges()` so I suppose it can apply to other operators as well. What do you think? I can reword.

Reply:
> [Add](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Add.html), [And](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$And.html), [Div](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Div.html), [Equal](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Equal.html), [Mod](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Mod.html), [Mul](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Mul.html), [NotEqual](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$NotEqual.html), [Or](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Or.html), [ShiftLeftByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$ShiftLeftByUInt.html), [ShiftLeftByUIntFixedWidth](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$ShiftLeftByUIntFixedWidth.html), [ShiftRightByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$ShiftRightByUInt.html), [Sub](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Sub.html), [Xor](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$BitVector$$Xor.html), [And](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$And.html), [Cat](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$Cat.html), [Equal](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$Equal.html), [NotEqual](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$NotEqual.html), [Or](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$Or.html), [ShiftLeftByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$ShiftLeftByUInt.html), [ShiftLeftByUIntFixedWidth](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$ShiftLeftByUIntFixedWidth.html), [ShiftRightByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$ShiftRightByUInt.html), [Xor](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$Bits$$Xor.html), [Add](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Add.html), [And](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$And.html), [Div](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Div.html), [Equal](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Equal.html), [Mod](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Mod.html), [Mul](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Mul.html), [NotEqual](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$NotEqual.html), [Or](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Or.html), [ShiftLeftByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$ShiftLeftByUInt.html), [ShiftLeftByUIntFixedWidth](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$ShiftLeftByUIntFixedWidth.html), [ShiftRightByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$ShiftRightByUInt.html), [Smaller](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Smaller.html), [SmallerOrEqual](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$SmallerOrEqual.html), [Sub](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Sub.html), [Xor](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$SInt$$Xor.html), [Add](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Add.html), [And](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$And.html), [Div](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Div.html), [Equal](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Equal.html), [Mod](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Mod.html), [Mul](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Mul.html), [NotEqual](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$NotEqual.html), [Or](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Or.html), [ShiftLeftByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$ShiftLeftByUInt.html), [ShiftLeftByUIntFixedWidth](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$ShiftLeftByUIntFixedWidth.html), [ShiftRightByUInt](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$ShiftRightByUInt.html), [Smaller](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Smaller.html), [SmallerOrEqual](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$SmallerOrEqual.html), [Sub](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Sub.html), [Xor](https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/Operator$$UInt$$Xor.html)
>
> -- https://spinalhdl.github.io/SpinalHDL/dev/spinal/core/internals/BinaryOperatorWidthableInputs.html

All operators working on sized signals? Or even: at places where a signal is expected with a given width?

Reply:
So we can leave it as it is?